sql/parser/eval_test.go

```diff
@@ -308,3 +315,25 @@ func TestEvalExprError(t *testing.T) {
 		}
 	}
 }
+
+func TestSimilarEscape(t *testing.T) {
+	testData := []struct {
+		expr     string
+		expected string
+	}{
+		{`test`, `test`},
+		{`test%`, `test.*`},
+		{`_test_`, `.test.`},
+		{`_%*`, `..**`},
+		{`[_%]*`, `[_%]*`},
+		{`.^`, `\.\^`},
+		{`%(b|d)%`, `.*(?:b|d).*`},
+		{`se\"arch\"[\"]`, `se(arch)[\"]`},
```

Comment:
Yeah it's a pretty strange syntax, although a lot of this half LIKE/half regexp syntax is kind of strange. It seems like that's what prompted Postrge to support standard [POSIX regexes](http://www.postgresql.org/docs/9.0/static/functions-matching.html#FUNCTIONS-POSIX-REGEXP) as well.

So by testing do you mean testing with PostgreSQL? If so, this is good for SIMILAR TO as well right?